src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -73,9 +73,13 @@ public void filterActiveAttachmentUnits(Lecture lectureWithAttachmentUnits) {
      * @param user                    the user for which this call should filter
      * @return lectures with filtered attachments
      */
-    public Set<Lecture> filterActiveAttachments(Set<Lecture> lecturesWithAttachments, User user) {
+    public Set<Lecture> filterVisibleLecturesWithActiveAttachments(Set<Lecture> lecturesWithAttachments, User user) {
         Set<Lecture> lecturesWithFilteredAttachments = new HashSet<>();
         for (Lecture lecture : lecturesWithAttachments) {
+            if (authCheckService.isOnlyStudentInCourse(lecture.getCourse(), user) && !lecture.isVisibleToStudents()) {
```

Comment:
Are all lectures here from the same course or can there be multiple courses? If it's only one course, consider moving the `isOnlyStudent` check out of the loop.

Reply:
They are always from the same course. I moved the check outside of the loop.